Source/JavaScriptCore/jsc.cpp

```diff
@@ -2521,11 +2521,21 @@ JSC_DEFINE_HOST_FUNCTION(functionSetTimeout, (JSGlobalObject* globalObject, Call
 
     // FIXME: We don't look at the timeout parameter because we don't have a schedule work later API.
     auto ticket = vm.deferredWorkTimer->addPendingWork(vm, callback, { });
-    vm.deferredWorkTimer->scheduleWorkSoon(ticket, [callback](DeferredWorkTimer::Ticket) {
-        JSGlobalObject* globalObject = callback->globalObject();
-        MarkedArgumentBuffer args;
-        call(globalObject, callback, jsUndefined(), args, "You shouldn't see this..."_s);
-    });
+
+    auto dispatch = [callback, ticket] {
+        callback->vm().deferredWorkTimer->scheduleWorkSoon(ticket, [callback](DeferredWorkTimer::Ticket) {
+            JSGlobalObject* globalObject = callback->globalObject();
+            MarkedArgumentBuffer args;
+            call(globalObject, callback, jsUndefined(), args, "You shouldn't see this..."_s);
+        });
+    };
+
+    JSValue timeout = callFrame->argument(0);
+    if (timeout.isNumber() && timeout.asNumber())
+        RunLoop::current().dispatchAfter(Seconds::fromMilliseconds(timeout.asNumber()), WTFMove(dispatch));
+    else
+        dispatch();
```

Comment:
What is the problem of always using `RunLoop` with `0_s`?

Reply:
I don't think there's any real issue. I just didn't because that would hit the runloop twice for no real reason.